examples/gen-go/pkg/typewriter/generated/main.go

```diff
@@ -0,0 +1,92 @@
+package typewriter
+
+import (
+	"encoding/json"
+	"github.com/segmentio/analytics-go"
+	"time"
+)
+
+type OrderCompleted struct {
+	Affiliation *string   `json:"affiliation,omitempty"` // Store or affiliation from which this transaction occurred (e.g. Google Store)
+	CheckoutID  *string   `json:"checkout_id,omitempty"` // Checkout ID
+	Coupon      *string   `json:"coupon,omitempty"`      // Transaction coupon redeemed with the transaction
+	Currency    *string   `json:"currency,omitempty"`    // Currency code associated with the transaction
+	Discount    *float64  `json:"discount,omitempty"`    // Total discount associated with the transaction
+	OrderID     string    `json:"order_id"`              // Order/transaction ID
+	Products    []Product `json:"products"`              // Products in the order
+	Revenue     *float64  `json:"revenue,omitempty"`     // Revenue associated with the transaction (excluding shipping and tax)
+	Shipping    *float64  `json:"shipping,omitempty"`    // Shipping cost associated with the transaction
+	Tax         *float64  `json:"tax,omitempty"`         // Total tax associated with the transaction
+	Total       *float64  `json:"total,omitempty"`       // Revenue with discounts and coupons added in. Note that our Google Analytics Ecommerce; destination accepts total or revenue, but not both. For better flexibility and total; control over tracking, we let you decide how to calculate how coupons and discounts are; applied
+}
```

Comment:
I'm wondering if the pointers are really necessary since with the use of omitempty, zero values (e.g. empty string) will be dropped from the json payload anyway?

Reply:
Hmm... nevermind, pointer types might be our only option here for props

Reply:
Yeah I'm not a fan either, but I don't think there's a better option.

Whenever we add the `omitempty` tag, then we have to make it a pointer. Otherwise, "zero values" like `0` will get dropped during serialization even if they should be included in the track call.

Though this brings up a good point that we can't do runtime required property validation unless we make required properties into pointers, too. Otherwise, if you (f.e.) had a required property, `discount`, and set it to `0` dollars then we would throw an error.

In that case, I'm going to make them all pointers.

Reply:
And to perform required property validation at run-time, we'll have to include `gojsonschema` as a dependency: https://github.com/xeipuuv/gojsonschema

We'll offer the same flag as `gen-js` to disable this validation code: `--runtimeValidation  Whether to output runtime validation code
                                                       [boolean] [default: true]`